### PR TITLE
Fixes #16861: Build agent on solaris

### PR DESCRIPTION
**Pull request by peckpeck**
https://issues.rudder.io/issues/16861

### CODE REVIEW

File: rudder-agent/SOURCES/Makefile [hunk 1]
```diff
@@ -607,6 +610,10 @@ build-cfengine-stamp:
 	cd cfengine-source && ./configure --prefix=$(RUDDER_DIR) --with-workdir=/var/rudder/cfengine-community --enable-static=yes --enable-shared=no --without-postgresql --without-mysql $(ACL_ARG) $(LMDB_ARG) $(OPENSSL_ARG) $(PCRE_ARG) $(CURL_ARG) $(YAML_ARG) $(XML_ARG) $(PAM_ARG) CFLAGS="$(CFLAGS) $(PIE_CFLAGS)" LDFLAGS="$(LDFLAGS)" CPPFLAGS="-I$(CURDIR)/dependencies$(RUDDER_DIR)/include/libxml2"
 	cd cfengine-source && $(MAKE) $(MAKE_J)
 	cd cfengine-source && $(MAKE) $(MAKE_J) install DESTDIR=$(CURDIR)/dependencies
+# solaris doesn't have uuidgen and needs one
+ifeq (solaris,$(OS_FAMILY))
+	gcc -luuid rudder-uuidgen.c -o rudder-uuidgen
```
Comment (amousset) [inline rudder-agent/SOURCES/Makefile]: don't we already have different uuid generation for AIX?

Comment (peckpeck) [reply rudder-agent/SOURCES/Makefile]: no, aix has an uuidgen binary in the system that can be called from the rudder-uuidgen script

**Comment by peckpeck** [comment]
PR rebased

File: rudder-agent/SOURCES/Makefile (hunk 1 above)
Comment (amousset) [reply rudder-agent/SOURCES/Makefile]: so we need to install into the destdir then?

Comment (peckpeck) [reply rudder-agent/SOURCES/Makefile]: in fact it has the same name as the script so it replaces it, so no need to change the install part

**Comment by peckpeck** [comment]
PR updated with a new commit

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
PR rebased

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
Commit modified

**Pull request merged**